ts/src/vector/flat/intFlatVector.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { describe, it, expect } from "vitest";
+import { IntFlatVector } from "./intFlatVector";
+import BitVector from "./bitVector";
+import { FlatSelectionVector } from "../filter/flatSelectionVector";
+
+function createVector(values: number[], name = "test"): IntFlatVector {
+    const data = new Int32Array(values);
+    return new IntFlatVector(name, data, values.length);
+}
+
+function createNullableVector(values: number[], nullBits: number, name = "test"): IntFlatVector {
+    const data = new Int32Array(values);
+    const nullability = new Uint8Array([nullBits]);
+    const bitVector = new BitVector(nullability, values.length);
+    return new IntFlatVector(name, data, bitVector);
+}
+
+function getSelectionIndices(selection: { selectionValues: () => number[]; limit: number }): number[] {
+    return selection.selectionValues().slice(0, selection.limit);
+}
+
+const simpleVector = createVector([10, 20, 30, 40, 50]);
+const withDuplicates = createVector([10, 20, 30, 20, 50, 10]);
+const withNulls = createNullableVector([10, 20, 30, 40, 50], 0b00010111);
+
+describe("IntFlatVector", () => {
+    describe("constructor and getValue", () => {
```

Comment:
The constructor is executed outside this `describe`, so I'm not sure this describes well the tests in this group...